### PR TITLE
slo as code

### PR DESCRIPTION
**Pull request by SudheendraKatte**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/service-reliability-management/slo/slo-as-code.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+## Prerequisites
+
+- A Git repo with at least one branch.​
+- A Git connector with a Personal Access Token (PAT) for your Git account.​
+
+The following sections explain how to set up Terraform based entities to manage SLO as code.
+
+
+## Setup Git Repository in GitHub
```
Comment (brian-f-harness) [inline docs/service-reliability-management/slo/slo-as-code.md:20]:
```suggestion
## Set up a Git Repository in GitHub
```

File: docs/service-reliability-management/slo/slo-as-code.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+8. Select Apply Changes to save and add the Terraform Plan step to the pipeline.
+
+
+## Add Terraform Apply step
+
+To add a Terraform Apply step:
+In the custom stage, select **Add Step** and choose **Terraform Apply**.
+Enter a name for the Terraform Apply step, select **Configuration Type**, and **Provisioner Identifier**.
+In **Configuration File Repository**, select the GitHub repository that you created.
+In the **Advanced** tab and select the delegate that you created as the default delegate for Terraform Plan.
```
Comment (brian-f-harness) [inline docs/service-reliability-management/slo/slo-as-code.md:67]:
```suggestion
On the **Advanced** tab, select the delegate that you created as the default delegate for Terraform Plan.
```

File: docs/service-reliability-management/slo/slo-as-code.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+Select **Apply Changes** to save and add the Terraform Plan step to the pipeline.
+
+## Create a webhook trigger
+To learn how to create a webhook trigger, go to [Trigger Deployments using Custom Triggers](https://developer.harness.io/docs/platform/triggers/trigger-deployments-using-custom-triggers/).
+
+
+## Add webhook to the GitHub repository
+
+To add webhook URL to GitHub repository:
```
Comment (brian-f-harness) [inline docs/service-reliability-management/slo/slo-as-code.md:76]:
```suggestion
To add a webhook URL to a GitHub repository:
```

File: docs/service-reliability-management/slo/slo-as-code.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+Enter a name for the Terraform Apply step, select **Configuration Type**, and **Provisioner Identifier**.
+In **Configuration File Repository**, select the GitHub repository that you created.
+In the **Advanced** tab and select the delegate that you created as the default delegate for Terraform Plan.
+Select **Apply Changes** to save and add the Terraform Plan step to the pipeline.
+
+## Create a webhook trigger
+To learn how to create a webhook trigger, go to [Trigger Deployments using Custom Triggers](https://developer.harness.io/docs/platform/triggers/trigger-deployments-using-custom-triggers/).
+
+
+## Add webhook to the GitHub repository
```
Comment (brian-f-harness) [inline docs/service-reliability-management/slo/slo-as-code.md:74]:
```suggestion
## Add a webhook to the GitHub repository
```

File: docs/service-reliability-management/slo/slo-as-code.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+To add webhook URL to GitHub repository:
+
+1. In your GitHub repository go to settings > webhooks.
+2. Add a new webhook and paste the payload URL from that you have copied from step 1.
```
Comment (brian-f-harness) [inline docs/service-reliability-management/slo/slo-as-code.md:79]:
```suggestion
2. Add a new webhook and paste the payload URL from that you copied in step 1.
```

File: docs/service-reliability-management/slo/slo-as-code.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+3. Paste the secret created while setting up the trigger.
+4. Select the event as **Just the push event** to trigger the webhook.
+5. Save the webhook.
+ 
+
+## Push changes to the repository
+
+Push changes to the GitHub repository, to trigger a workflow for the pipeline. During the pipeline run, the Terraform Plan step fetches changes from GitHub and applies to all the entities which have changes. The changes can be creating a new entity, updating an existing entity, or deleting an existing entity. During the pipeline run, the Terraform Plan step fetches the latest changes from the repository and generates a plan for updating the infrastructure. The plan may include creating a new entity, updating an existing entity, or deleting an existing entity.
```
Comment (brian-f-harness) [inline docs/service-reliability-management/slo/slo-as-code.md:87]:
```suggestion
Push your changes to the GitHub repository to trigger a workflow for the pipeline. During the pipeline run, the Terraform Plan step fetches changes from GitHub and applies them to all updated entities. The changes can be creating a new entity, updating an existing entity, or deleting an existing entity. During the pipeline run, the Terraform Plan step fetches the latest changes from the repository and generates a plan for updating the infrastructure. The plan may include creating a new entity, updating an existing entity, or deleting an existing entity. 
```

**Review by brian-f-harness — DISMISSED** [review]
lgtm. Added a couple suggestions.

**Comment by SudheendraKatte** [comment]
Already peer reviewed and approved.

**Pull request merged**